### PR TITLE
Update Admin API

### PR DESCRIPTION
**Pull request by dnfd**
- [x] Update RansackBuilder
- [ ] Update entities desc (optional)
- [x] Remove commented code in controllers
- [ ] Inspect code for n + 1 
- [x] Entities can inherit only from `API::V2::Entities` otherwise define new entity
- [x] Update params according to next changes https://github.com/rubykube/peatio/commit/283f7dd7c5291b91f46c34cb1b90c9c928438407
- [x] Avoid adding from & to params implicitly

### CODE REVIEW

File: app/api/v2/admin/withdraws.rb [hunk 1]
```diff
@@ -45,70 +41,16 @@ class Withdraws < Grape::API
 
           ransack_params = Helpers::RansackBuilder.new(params)
                              .eq(:id, :txid, :rid, :tid)
-                             .map(aasm_state: :state, member_uid: :uid, account_id: :account, currencie_id: :currency)
-                             .build(type_eq: params[:type].present? ? "Withdraws::#{params[:type]}" : nil)
+                             .translate(:state => :aasm_state, :uid => :member_uid, :account => :account_id, :currency => :currency_id)
```
Comment (ysv) [inline app/api/v2/admin/withdraws.rb]: I think preferred syntax is `key: :value`

File: app/api/v2/admin/orders.rb [hunk 2]
```diff
@@ -43,11 +43,12 @@ class Orders < Grape::API
 
           ransack_params = Helpers::RansackBuilder.new(params)
                              .eq(:price, :origin_volume, :ord_type)
-                             .map(market_id: :market, member_uid: :uid, member_email: :email)
-                             .build({
+                             .translate(:market => :market_id, :uid => :member_uid, :email => :member_email)
```
Comment (ysv) [inline app/api/v2/admin/orders.rb]: I think preferred syntax is `key: :value`

File: app/api/v2/admin/helpers.rb [hunk 3]
```diff
@@ -10,24 +10,30 @@ module Helpers
...
+
+          def with_daterange
+            @build.merge!("#{@params[:range]}_at_gteq" => Time.at(@params[:from])) if @params[:from]
+            @build.merge!("#{@params[:range]}_at_lt" => Time.at(@params[:to])) if @params[:to]
+            self
           end
 
-          def map(opt)
-            opt.each { |k, v| @build.merge!("#{k}_eq" => @params[v]) }
+          def translate(opt)
```
Comment (ysv) [inline app/api/v2/admin/helpers.rb:35]: as for me it's better to make this translate without appending *_eq*
so we use it 
```ruby
.translate(state: :aasm_state_eq)
```

**Review by ysv — COMMENTED** [review]
Copy TODO to this PR pls and update it

**Comment by pkucherenk0** [comment]
1. Blockchains/new endpoint allowing as to create blockchain with empty server
2. Blockchains/new and /update missing validation on server parameter
3. Blockchains/new trying to create new blockchain that have same client with already existed blockchain will course an server.internal_error

**Comment by pkucherenk0** [comment]
1. Currencies/update & /new create validation for base_factor, it can only be in form of "1" + N amount of "0" and limited with 19 numbers in total. (1; 10; 100; 1000; 1000; 10000; 1+n; ...; 1000000000000000000). 
2. Currencies/update & /new precision (&position) cant be negative number.
3. Currencies/update & /new min_deposit_amount validation error is confusing ```admin.currency.invalid_deposit_fee```

File: app/api/v2/admin/helpers.rb (hunk 3 above)
Comment (dnfd) [reply app/api/v2/admin/helpers.rb:35]: Why it is better?

**Pull request merged**